core/src/main/java/org/jboss/gwt/elemento/core/BodyObserver.java

```diff
@@ -0,0 +1,149 @@
+package org.jboss.gwt.elemento.core;
+
+import elemental2.dom.*;
+import jsinterop.base.Js;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static elemental2.dom.DomGlobal.document;
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+class BodyObserver {
+
+    private static String ATTACH_UID_KEY = "on-attach-uid";
+    private static String DETACH_UID_KEY = "on-detach-uid";
+
+    private static List<ElementObserver> detachObservers = new ArrayList<>();
+    private static List<ElementObserver> attachObservers = new ArrayList<>();
+
+    static {
+        MutationObserver mutationObserver = new MutationObserver((MutationRecord[] records, MutationObserver observer) -> {
+            for (int i = 0; i < records.length; i++) {
+                onElementsRemoved(records[i]);
+                onElementsAppended(records[i]);
+            }
+            return null;
+        });
+
+        MutationObserverInit mutationObserverInit = MutationObserverInit.create();
+        mutationObserverInit.setChildList(true);
+        mutationObserverInit.setSubtree(true);
+
+        mutationObserver.observe(document.body, mutationObserverInit);
```

Comment:
Does this mean `onElementsRemoved()` and `onElementsAppended()` will be called for *all* elements removed and added to the DOM from this moment on?

Reply:
yes, but its not a call for each single dom add/remove. E.g. if you add `<div><ul><li/><li/></ul><div>` there will be 1 event, not 4 events.

more infos, see here: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver

Reply:
yes, but its not a call for each single dom add/remove. E.g. if you add `<div><ul><li/><li/></ul><div>` there will be 1 event, not 4 events.

more infos, see here: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver